### PR TITLE
Fix clear_cache function

### LINKED ISSUE #19: Use `wpdb` to remove all Head Tags transients
Currently, an option is being used to store the name of all transients. The list of transient names could be huge and this doesn't scale well. We should use `wpdb` to remove the transients based on a prefix instead of store all the names.

### PR DESCRIPTION
**Pull request by DAreRodz**
This PR finish the work that was done in #23.

Fixes #19

### CODE REVIEW

**Comment by DAreRodz** [comment]
Now the `clear_cache` function is taking too long. 🙁

I have an idea, though. We can, instead of removing all transients, invalidate them using a validation _token_ or something similar.

That _token_ would be a value generated each time you click an _invalidate_ button and stored in the database.

When a head tags transient is updated or generated for the first time, it saves the current _token_ along with the head tags generated, in the same transient (or in other transient).

Then, when a head tags transient is gotten (using `get_transient`), its _token_ is compared with the current _token_ value generated by the button mentioned before. If it's the same, it returns the head tags. If not, it generate them again and update its _token_.

That way we wouldn't have to iterate over all of the transients and remove them (which is too heavy) and they would update on-demand as well.

What do you think, @luisherranz?

---

PS: Whe can leave the _purge cache_ button too, just in case.

**Comment by DAreRodz** [comment]
Using a cache token is working as expected. 🙌

The time it takes to generate the head tags is almost the same, and invalidate the cache is just one click, so it scales very well.

Now I have to change the tests. Working on it right now.

**Comment by DAreRodz** [comment]
I'm considering to remove the `clear_cache` function and just leave the invalidation.

**Comment by DAreRodz** [comment]
Thanks for the detailed review, @luisherranz!

I think you are right. It seems like what I did was just to find a solution to a problem caused by using transients. I'm going to investigate if using the WordPress' meta tables is a better option (I need to figure out how because I never used them).

About your comments:

> It feels weird for me to generate the token in the client.

I decided to use the `frontity_headtags_settings` to store that token in the database and that settings come from the frontend.

But, yes, It's weird for me too. The server should be responsible of doing this.

> We must delete the transients in the uninstall method. We cannot leave those hanging around.

We could do that, but not using `delete_transient`. I explain below why.

> And, is there a reason not to delete the transients on the purge/invalidation?

The reason is it would take a loooong time to do so, in the case you were using `delete_transient`, or may derive into cache inconsistency if you use a `DELETE` query directly.

**Comment by luisherranz** [comment]
> The reason is it would take a loooong time to do so, in the case you were using delete_transient, or may derive into cache inconsistency if you use a DELETE query directly.

You've solved the cache inconsistency with the token so you can safely delete the transients using `DELETE`.

**Comment by luisherranz** [comment]
> I'm going to investigate if using the WordPress' meta tables is a better option (I need to figure out how because I never used them).

Neither did I. Don't worry, we'll gain more experience as we go :relaxed:

**Comment by DAreRodz** [comment]
> You've solved the cache inconsistency with the token so you can safely delete the transients using DELETE.

Oh, I see. The head tags could be in the cache but at least our plugin would know they are not valid anymore.

**Comment by DAreRodz** [comment]
Not adding a Changeset file because it was already added in the previous PR (#23). It's [this one](https://github.com/frontity/wp-plugins/blob/fix-clear-cache-better/.changeset/dirty-fans-divide.md).

File: plugins/frontity-headtags/admin/config.js [hunk 1]
```diff
@@ -45,11 +39,14 @@ export default {
         state.headtags.settings.isEnabled = false;
         actions.headtags.save();
       },
-      clearCache: async ({ state, actions }) => {
-        console.log("Invalidating cache");
-        state.headtags.settings.cacheToken = generateToken();
-        await actions.headtags.save();
-        console.log("cache is invalid now");
+      clearCache: async ({ state }) => {
+        console.log("clearing cache");
```
[the remark this answers is not in the document]
Comment (DAreRodz) [reply plugins/frontity-headtags/admin/config.js]: Oops, I forgot those `console.log`s. Removing them before merging.

**Pull request merged**